.github/workflows/fuzz.yaml

```diff
@@ -0,0 +1,58 @@
+name: Fuzzing
+
+on:
+  pull_request:
+    types: [labeled]
+  schedule:
+    - cron: "14 3 * * *"
+
+jobs:
+  fuzz-matrix:
+    if: ${{ github.event.label.name == 'action/fuzz' }}
+    runs-on: ubuntu-latest
+    outputs:
+      fuzz-names: ${{ steps.list-fuzz.outputs.list }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: '1.18'
+      - id: list-fuzz
+        run: |
+          list="$(go test -list '^Fuzz' ./... | \
+                  grep '^Fuzz' | \
+                  xargs jq -n '$ARGS.positional' --compact-output --args)"
+          echo "::set-output name=list::${list}"
+
+  fuzz:
+    needs: fuzz-matrix
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        name: ${{ fromJson(needs.fuzz-matrix.outputs.fuzz-names) }}
+    env:
+      GOCACHE: /tmp/go/cache
+    steps:
+      - run: mkdir -p ${GOCACHE}
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: '1.18'
+
+      - uses: actions/cache@v3
+        with:
+          key: go-fuzz-corpus-${{ matrix.name }}-${{ github.run_number }}
+          restore-keys: go-fuzz-corpus-${{ matrix.name }}-
+          path: /tmp/go/cache/fuzz
```

Comment:
Looking at https://github.com/infrahq/infra/runs/6430042316 (the most recent run on this PR), it seems like the cache is being saved, but the restore isn't working for some reason. This `restore-keys` is a prefix match, I'm not sure why it's not matching.

Reply:
I read over https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#matching-a-cache-key again, and it seems like this should be correct.

I'm not sure if I'm missing something, or if this config needs to be in main first.

Reply:
This looks right to me. The run even says it cached successfully

Reply:
Perhaps the run number is causing problems. wouldn't that make each of the caches unique, so you could never restore it?

Reply:
Ya, the `run_number` is unique for each run.  `restore-keys` will do a prefix match though, which should still be a match on previous runs. The docs for that are here: https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#matching-a-cache-key

I'm guessing it's because this config isn't in `main` yet, so something about the [restrictions](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#restrictions-for-accessing-a-cache) is preventing a cache hit.